src/stores/nodeParamStore.ts

```diff
@@ -0,0 +1,273 @@
+// should apply this to litegraph node
+import { defineStore } from 'pinia'
+import { ref, computed, watch } from 'vue'
+import { app } from '@/scripts/app'
+
+// todo: Use Litegraph.js types here when we have a Litegraph ts version or Zod Schema
+interface Widget {
+  title: string
+  type: string
+  label: string
+  options: any
+  value: object | string | number | boolean | Array<any> | null
+  fav: boolean
+}
+
+interface Node {
+  title: string
+  id: number
+  order: number
+  color: string
+  bypass: boolean
+  bgcolor: string
+  mode: number
+  favWidgets: FavWidgets[]
+  category: string
+  widgets: Widget[]
+}
+
+interface FavWidgets {
+  name: string
+  fav: boolean
+}
+
+// interface Gallery {
+//     images: string[];
+// }
+
+interface category {
+  name: string
+  count: number
+}
+
+export const useNodeParamStore = defineStore('nodeParam', () => {
+  // State
+  const nodes = ref<Node[]>([])
+  const widgetFav = ref<Widget[]>([])
+  let filterNodes = ref(nodes)
+  const categories = computed<category[]>(() => {
+    let favCount = 0
+    for (const node of nodes.value) {
+      for (const widget of node.widgets)
+        if (widget.fav) {
+          favCount++
+        }
+    }
+    return [
+      { name: 'All', count: nodes.value.length },
+      { name: 'Star', count: favCount },
+      {
+        name: 'Notnull',
+        count: nodes.value.filter((node) => node.widgets.length > 0).length
+      }
+    ]
+  })
+  // const gallery = ref<Gallery>({ images: [] });
+
+  let isEditing = false
+
+  //Setters
+  const setIsEditing = (value: boolean) => {
+    isEditing = value
+  }
+  // Actions
+  const updateNodes = () => {
+    // @ts-expect-error app.graph._nodes is private, there's no export functions to access it
+    const graphNodes = ref([...app.graph._nodes])
+
+    //todo: remove this when app.graph._nodes supports ref()
```

Comment:
It seems like we are using a very awkward method to sync with litegraph state. I think it is fine for now. However do you have any design on how should make litegraph objects reactive, while maintaining the performance? (For example, we probably don't want to make node position reactive, as that would cause lag when dragging the node)

Reply:
Possible method:
1. easiest way: add callback on node/widget value change, without rewrite the litegraph.js
2. Add a get() set() to litegraph objects, which should be a big rewrite task which may cause unpredictive bug
Love to hear your advice